brambling/views/utils.py

```diff
@@ -225,3 +228,77 @@ def get_context_data(self, **kwargs):
             'next_step': self.current_step.next_step if self.current_step else None,
         })
         return context
+
+
+class FinanceTable(object):
+
+    def __init__(self, event, transactions):
+        self.event = event
+        self.transactions = transactions
+
+    def headers(self):
+        return [
+            'Timestamp (%s)' % self.event.timezone,
+            'Created By',
+            'Method',
+            'Type',
+            'Order',
+            'Amount',
+            'Dancerfly Fee',
+            'Payment Fee',
+        ]
+
+    def header_row(self):
+        return [self.headers()]
+
+    def plaintext_rows(self):
+        yield self.headers()
+        for transaction in self.transactions:
+            yield [c.value for c in self.format_transaction(transaction)]
+
+    def cell_rows(self):
+        for transaction in self.transactions:
+            yield self.format_transaction(transaction)
+
+    def transaction_rows(self):
+        return islice(self.all_rows(), 1, None)
+
+    def transaction_as_cell_row(self, transaction):
+        return self.format_transaction(transaction)
+
+    def money(self, amount):
+        return format_money(amount, self.event.currency)
+
+    def format_transaction(self, transaction):
+        return [
+            Cell(field='timestamp', value=self.format_timestamp(transaction)),
+            Cell(field='created_by', value=self.created_by_name(transaction)),
+            Cell(field='method', value=transaction.get_method_display()),
+            Cell(field='type',
+                 value=transaction.get_transaction_type_display()),
+            Cell(field='order', value=self.order_code(transaction)),
+            Cell(field='amount', value=self.money(transaction.amount)),
+            Cell(field='application_fee',
+                 value=self.money(transaction.application_fee)),
+            Cell(field='processing_fee',
+                 value=self.money(transaction.processing_fee)),
+        ]
+
+    def format_timestamp(self, transaction):
+        tz = pytz.timezone(self.event.timezone)
+        localtime = timezone.localtime(transaction.timestamp, timezone=tz)
+        return localtime.strftime("%B %d, %Y %H:%M:%S")
+
+    @classmethod
```

Comment:
It would be more in keeping with Django's general behavior – admin, feeds, etc – (and the general behavior of Dancerfly) to switch these to instance methods rather than class methods.